--- mod_encoding.c	2008-05-16 09:40:48.000000000 +0200
+++ mod_encoding.c.oden	2008-05-16 09:40:37.000000000 +0200
@@ -74,9 +74,9 @@
 #include <http_log.h>
 #include <http_protocol.h>
 #include <http_request.h>
-/* #include <ap_compat.h> */
+#include <apr_strings.h>
 
-#include <iconv_hook/iconv.h>
+#include <iconv.h>
 
 /**
  * Core part of the module.
@@ -232,7 +232,7 @@
 
 
   for (i = 0 ; i < encmap->nelts ; i += 2) {
-    if (ap_regexec((regex_t *)list[i], lookup, 0, NULL, 0) == 0) {
+    if (ap_regexec((ap_regex_t *)list[i], lookup, 0, NULL, 0) == 0) {
       apr_array_cat(encs, (apr_array_header_t *)list[i + 1]);
       return encs;
     }
@@ -290,7 +290,7 @@
   /* register useragent with UserAgent: pattern */
   if (*args && (arg = ap_getword_conf_nc(cmd->pool, &args))) {
     *(void **)apr_array_push(conf->client_encoding) =
-      ap_pregcomp(cmd->pool, arg, REG_EXTENDED|REG_ICASE|REG_NOSUB);
+      ap_pregcomp(cmd->pool, arg, AP_REG_EXTENDED|AP_REG_ICASE|AP_REG_NOSUB);
   }
 
   /* register list of possible encodings from above useragent */
